draft-kazuho-httpbis-priority.md

```diff
@@ -112,33 +112,93 @@ Unknown parameters MUST be ignored.
 
 ## urgency
 
-The `urgency` parameter takes one of the following sh-tokens as the value that
-indicates how an HTTP response affects the usage of other responses:
+The `urgency` parameter takes an integer between -1 and 6 as shown below:
 
-* `blocking` indicates that the response prevents other responses from being
-  used.
-* `document` indicates that the response contains the document that is being
-  processed.
-* `non-blocking` indicates that the response does not prevent the client from
-  using the document even though the response is being used or referred to by
-  the document.
+| Urgency         | Definition                      |
+|----------------:|:--------------------------------|
+|              -1 | blocking ({{blocking}})         |
+|               0 | document ({{document}})         |
+| between 1 and 5 | non-blocking ({{non-blocking}}) |
+|               6 | background ({{background}})     |
+{: #urgencies title="Urgencies"}
 
-The default value is `document`.
+The value is encoded as an sh-integer.  The default value is zero.
 
-A server SHOULD transmit HTTP responses in the order of their urgency:
-`blocking` first, followed by `document`, followed by `non-blocking`.
+A server SHOULD transmit HTTP responses in the order of their urgency values.
+The lesser the value is, the higher the precedence is.
 
 The following example shows a request for a CSS file with the urgency set to
-`blocking`:
+`-1`:
 
 ~~~ example
 :method = GET
 :scheme = https
 :authority = example.net
 :path = /style.css
-priority = urgency=blocking
+priority = urgency=-1
 ~~~
 
+The definition of the urgencies and their expected use-case are described below.
+Endpoints SHOULD respect the definition of the values when assigning urgencies.
+
+### blocking
+
+The blocking urgency indicates that the response prevents other responses from
+being used.
+
+Value of -1 is used to represent this urgency.
+
+For example, a stylesheet being referred from the HEAD section of an HTML
+document blocks a web browser from rendering the HTML.  In such case, the
+stylesheet is given the blocking urgency.
+
+### document
+
+The document urgency indicates that the response contains the document that is
+being processed.  This urgency is also used for responses that deserve the same
+precedence as the contents of the document.
+
+Value of zero (0) is used to represent this urgency.
+
+For example, when a user using a web browser navigates to a different HTML
+document, the request for that HTML is given the document urgency.  When that
+HTML document uses a custom font, the request for that custom font SHOULD also
+be given the document urgency.  This is because the availablity of the custom
+font is the precondition for the user to use the portion of the HTML document,
```

Comment:
> is the precondition

is likely a precondition
(because nowadays you have things like font-display, which allows developers to specify a fallback)

> to use the portion of the HTML

to use that portion of the HTML

Reply:
Fixed in 4e0e79e.